### PR TITLE
Adding close/open dome to the states

### PR DESCRIPTION
**Pull request by bazkiaei**
The `open_dome` and `close_dome` command are added to the states of `huntsman-pocs`. These commands currently control the shutter and not the dome. This PR regards issue #237 but does not cover all items of the issue.

### CODE REVIEW

**Comment by lspitler** [comment]
@bazkiaei can you check if there are unit tests associated with the states? If so, see if you need to add/modify a unit test from this edit.

**Comment by lspitler** [comment]
Also @bazkiaei there seems to be an error that reflects a change in your code. Can you investigate?

File: src/huntsman/pocs/states/huntsman/parking.py
Comment (danjampro) [inline src/huntsman/pocs/states/huntsman/parking.py]: This isn't necessarily true, for instance if the observatory has no dome. This may cause some confusion in the logs.

File: src/huntsman/pocs/states/huntsman/parking.py
Comment (danjampro) [inline src/huntsman/pocs/states/huntsman/parking.py]: This line is unnecessary.

File: src/huntsman/pocs/states/huntsman/ready.py
Comment (danjampro) [inline src/huntsman/pocs/states/huntsman/ready.py]: As above.

File: src/huntsman/pocs/states/huntsman/ready.py
Comment (danjampro) [inline src/huntsman/pocs/states/huntsman/ready.py]: As above.

File: src/huntsman/pocs/tests/test_states.py
Comment (danjampro) [inline src/huntsman/pocs/tests/test_states.py:52]: I'm not sure why we are using `set_dome` here. Surely we want to pass it to the observatory instance on creation like `HuntsmanObservatory(dome=dome, ...)` ?

File: src/huntsman/pocs/tests/test_states.py
Comment (danjampro) [inline src/huntsman/pocs/tests/test_states.py:108]: It would be good to make sure the dome is actually open to begin with here.

e.g. `pocs.observatory.dome.open()` and/or `assert pocs.observatory.dome.is_open`

File: src/huntsman/pocs/tests/test_states.py
Comment (danjampro) [inline src/huntsman/pocs/tests/test_states.py]: It would be good to make sure the dome is not already closed here.

**Review by danjampro — COMMENTED** [review]
@bazkiaei looking good! A few minor comments to address before I approve.

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py:52]: I think this is how it should work, but it does not:
```
pocs = POCS(observatory, run_once=True, dome=dome)
```
So I did it that way.

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py:108]: The command `pocs.get_ready()` in the next line will open the dome. If it does not, the next test will raise an error!

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py]: The aim of this test here is to make sure that the state `ready` can open the dome. So it is important to close the dome before executing the state. This is the reason I put `pocs.observatory.dome.close()` in the next line. I believe the latest method should be able to manage the situation if the dome is already closed (and keep it closed). Does this address your comment?

File: src/huntsman/pocs/states/huntsman/parking.py
Comment (bazkiaei) [reply src/huntsman/pocs/states/huntsman/parking.py]: What do you think about the following instead:
```
if pocs.observatory.has_dome:
    pocs.say("I'm opening the dome.")
```

File: src/huntsman/pocs/tests/test_states.py
Comment (lspitler) [reply src/huntsman/pocs/tests/test_states.py:52]: Why didn't it work? What is the error? @bazkiaei

File: src/huntsman/pocs/tests/test_states.py
Comment (lspitler) [reply src/huntsman/pocs/tests/test_states.py:108]: @danjampro what scenario are you trying to catch here? In any case, it wouldn't hurt to add these checks.

File: src/huntsman/pocs/states/huntsman/parking.py
Comment (lspitler) [reply src/huntsman/pocs/states/huntsman/parking.py]: Sounds good. You should do this everywhere you "say" open/close the dome.

File: src/huntsman/pocs/states/huntsman/ready.py
Comment (lspitler) [inline src/huntsman/pocs/states/huntsman/ready.py]:
```suggestion
            pocs.logger.warning('Not opening the dome, Observatory has no dome attribute.')
```

File: src/huntsman/pocs/states/huntsman/parking.py
Comment (lspitler) [reply src/huntsman/pocs/states/huntsman/parking.py]: Agree @bazkiaei . Remove these "Excepting" logger calls.

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py:52]: It looks like it does not create the dome attribute that way. When I run the test and try to check the dome status, it gives the following error:

```
   assert not pocs.observatory.dome.is_open
E AttributeError: 'NoneType' object has no attribute 'is_open'
```

Comment (lspitler) [reply src/huntsman/pocs/tests/test_states.py:52]: @bazkiaei I think that implies something else is going wrong with the dome simulator creation. Can you verify `create_dome_simulator` is working here?

**Comment by lspitler** [comment]
[POCS dome shutter control](https://trello.com/c/6GJsZa8w/178-pocs-dome-shutter-control)

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py:52]: @lspitler I checked `create_dome_simulator` and it is working here. The output of that is `class 'panoptes.pocs.dome.simulator.Dome'`, which is what we want. Following @danjampro advice, I keep `pocs.observatory.set_dome(dome)` and open an issue in POCS.

File: src/huntsman/pocs/tests/test_states.py
Comment (bazkiaei) [reply src/huntsman/pocs/tests/test_states.py:108]: To prevent repetition in tests, I merge the dome `open` and `close` states tests in one test.

File: src/huntsman/pocs/tests/test_states.py
Comment (lspitler) [reply src/huntsman/pocs/tests/test_states.py:52]: sounds good @bazkiaei

**Comment by lspitler** [comment]
@bazkiaei would you like @danjampro or I to do a final review?

**Comment by bazkiaei** [comment]
> @bazkiaei would you like @danjampro or I to do a final review?

I will appreciate if you do it @lspitler .

File: src/huntsman/pocs/tests/test_states.py
Comment (lspitler) [reply src/huntsman/pocs/tests/test_states.py:52]: Can you make a new in the POCS issue linking to this PR? Want to probably change the call here once POCS is fixed.

**Review by lspitler — APPROVED** [review]
Looks good @bazkiaei . You can merge after you test on Huntsman.

**Comment by bazkiaei** [comment]
@lspitler I already had opened an issue regarding this problem. I added this PR link to that issue. Does it work?
And here is the issue that I opened:
 [https://github.com/panoptes/POCS/issues/1042](url)

**Review by danjampro — APPROVED** [review]
@bazkiaei Looks good to me. 

@lspitler It will probably be easier to merge this now rather than test on huntsman before, as we would otherwise need to mess around with the docker setup.

**Comment by lspitler** [comment]
@danjampro makes sense. Will merge after tests done... or go ahead and do so yoursef.

@bazkiaei if after testing on the live huntsman you find you need to make changes, I think you can re-open this PR and submit code changes here. Keeps relevant edits in 1 spot.

**Pull request merged**
